public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -109,16 +110,20 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
     }
 
     if (targets.nativeSearch?.length) {
-      const searchQuery = this.buildSearchQuery(targets.nativeSearch[0]);
-      subQueries.push(
-        this._request('/api/search', searchQuery).pipe(
-          map((response) => {
-            return {
-              data: [createTableFrameFromSearch(response.data.traces, this.instanceSettings)],
-            };
-          })
-        )
-      );
+      try {
+        const searchQuery = this.buildSearchQuery(targets.nativeSearch[0]);
+        subQueries.push(
+          this._request('/api/search', searchQuery).pipe(
+            map((response) => {
+              return {
+                data: [createTableFrameFromSearch(response.data.traces, this.instanceSettings)],
+              };
+            })
+          )
+        );
+      } catch (error) {
+        return of({ error: { message: error.message }, data: [] });
+      }
```

Comment:
We can use observable's `catchError` here, instead of wrapping it in try-catch blog. Also then when I console.log-ed the format of the error and where to find message, it was in `error.data.message`

```
    if (targets.nativeSearch?.length) {
      const searchQuery = this.buildSearchQuery(targets.nativeSearch[0]);
      subQueries.push(
        this._request('/api/search', searchQuery).pipe(
          map((response) => {
            return {
              data: [createTableFrameFromSearch(response.data.traces, this.instanceSettings)],
            };
          }),
          catchError((error) => {
            return of({ error: { message: error.data.message }, data: [] });
          })
        )
      );
    }
```

Reply:
`buildSearchQuery` will throw an error if the validation fails, so I need to keep the trycatch. I'll add the `catchError` to the request pipe to handle that which will work nicely because the error messages are on different fields like you caught.